### PR TITLE
fix #305134: crash on chordsymbol playback

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**

### CODE REVIEW

File: libmscore/part.cpp [hunk 1]
```diff
@@ -598,7 +598,7 @@ void Part::updateHarmonyChannels(bool isDoOnInstrumentChanged, bool checkRemoval
       if (!harmonyChannel() && harmonyCount() > 0) {
             Instrument* instr = instrument();
             Channel* c = new Channel(*instr->channel(0));
-            c->setName(qApp->translate("channel", Channel::HARMONY_NAME));
+            c->setName(Channel::HARMONY_NAME);
```
Comment (Jojo-Schmitz) [inline libmscore/part.cpp:601]: This is what causes translated names to get scored in the score file and subsequently causes the crash at hand in this issue

File: libmscore/instrument.cpp [hunk 2]
```diff
@@ -26,8 +26,8 @@
 
 namespace Ms {
 
-const char* Channel::DEFAULT_NAME = QT_TRANSLATE_NOOP("channel", "Normal");
-const char* Channel::HARMONY_NAME = QT_TRANSLATE_NOOP("channel", "Chord symbols");
+const char* Channel::DEFAULT_NAME = QT_TRANSLATE_NOOP("InstrumentsXML", "Normal");
+const char* Channel::HARMONY_NAME = QT_TRANSLATE_NOOP("InstrumentsXML", "Chord symbols");
```
Comment (Jojo-Schmitz) [inline libmscore/instrument.cpp:30]: Using the same context as the rest of the instruments, long- and short names, channels, etc..
Simplifies the code that picks up those translations quite a but, but means that the current translations won't catch it (as they  have a different context), so needs to get merged and translated to be able to see whether it then shows the translated names in the UI.

Comment (Jojo-Schmitz) [reply libmscore/instrument.cpp:30]: Wonder whether to use `QT_TRANSLATE_NOOP("InstrumentsXML", "Normal"`, "Channel")` to give some hint to the translator?

File: libmscore/part.cpp [hunk 3]
```diff
@@ -610,17 +610,17 @@ void Part::updateHarmonyChannels(bool isDoOnInstrumentChanged, bool checkRemoval
...
+            int chanIdx = instr->channelIdx(Channel::HARMONY_NAME);
+            if (chanIdx == -1)
+                  return nullptr;
 
-           const Channel* chan = instr->channel(chanIdx);
-           Q_ASSERT(chan);
-           return chan;
+            const Channel* chan = instr->channel(chanIdx);
+            Q_ASSERT(chan);
+            return chan;
```
Comment (Jojo-Schmitz) [inline libmscore/part.cpp:623]: plain formatting changes, in the passing

File: mscore/stafftextproperties.cpp [hunk 4]
```diff
@@ -184,10 +184,10 @@ StaffTextProperties::StaffTextProperties(const StaffTextBase* st, QWidget* paren
             const Channel* a = part->instrument(tick)->channel(i);
             QTreeWidgetItem* item = new QTreeWidgetItem(channelList);
             item->setData(0, Qt::UserRole, i);
-            if (a->name().isEmpty() || a->name() == Channel::DEFAULT_NAME)
-                  item->setText(0, qApp->translate("channel", Channel::DEFAULT_NAME));
-            else
-                  item->setText(0, qApp->translate("InstrumentsXML", a->name().toUtf8().data()));
+            QString name = a->name();
+            if (a->name().isEmpty())
+                  name == Channel::DEFAULT_NAME;
```
Comment (mattmcclinch) [inline mscore/stafftextproperties.cpp:189]: This is supposed to be assignment, right?

File: mscore/stafftextproperties.cpp [hunk 5]
```diff
@@ -388,26 +388,20 @@ void StaffTextProperties::channelItemChanged(QTreeWidgetItem* item, QTreeWidgetI
...
-                  ti->setText(0, qApp->translate("channel", Channel::DEFAULT_NAME));
-                  ti->setData(0, Qt::UserRole, Channel::DEFAULT_NAME);
-                  }
-            else {
-                  ti->setText(0, qApp->translate("InstrumentsXML", e.name.toUtf8().data()));
-                  ti->setData(0, Qt::UserRole, e.name);
-                  }
+            QString name = e.name;
+            if (e.name.isEmpty())
+                  name == Channel::DEFAULT_NAME;
```
Comment (mattmcclinch) [inline mscore/stafftextproperties.cpp:393]: And here?

File: mscore/stafftextproperties.cpp [hunk 6]
```diff
@@ -388,26 +388,20 @@ void StaffTextProperties::channelItemChanged(QTreeWidgetItem* item, QTreeWidgetI
...
```
Comment (mattmcclinch) [inline mscore/stafftextproperties.cpp:402]: And here?

File: mscore/stafftextproperties.cpp (hunk 4 above)
Comment (Jojo-Schmitz) [reply mscore/stafftextproperties.cpp:189]: Outch... Of course

**Comment by Jojo-Schmitz** [comment]
Ok that is pretty bad. Thanks not noticing! Something to fix, tomorrow...

**Comment by Jojo-Schmitz** [comment]
See #6082

**Pull request merged**